src/main/core/Repository/User/UserRepository.php

```diff
@@ -545,4 +545,19 @@ public function findAllEnabledUsers($executeQuery = true)
 
         return $executeQuery ? $query->getResult() : $query;
     }
+
+    public function findUserByNameOnFailure($username)
+    {
+        $dql = '
+            SELECT u FROM Claroline\CoreBundle\Entity\User u
+            WHERE u.username = :username
+            OR u.email = :username
+            AND u.isEnabled = true
+        ';
+
+        $query = $this->_em->createQuery($dql);
+        $query->setParameter('username', $username);
+
+        return $query->getOneOrNullResult();
+    }
```

Comment:
You should use `findByName` instead of creating a new method.
Also note you will not be able to retrieve the user with this method if there is an authentication failure because the user is disabled.

Reply:
I removed the function to use findByName instead, thx :)